lua/bufferline/layout.lua

```diff
@@ -2,19 +2,15 @@
 -- layout.lua
 --
 
-local vim = vim
-local nvim = require'bufferline.nvim'
-local utils = require'bufferline.utils'
 local Buffer = require'bufferline.buffer'
-local len = utils.len
-local strwidth = nvim.strwidth
-
+local strwidth = vim.api.nvim_strwidth
+local utils = require'bufferline.utils'
 
 local SIDES_OF_BUFFER = 2
 
 local function calculate_tabpages_width()
   local current = vim.fn.tabpagenr()
-  local total   = vim.fn.tabpagenr('$')
+  local total   = #vim.api.nvim_list_tabpages()
```

Comment:
TODO: I figured running an API function would be faster than running a VimScript function but I should actually test this.

Reply:
Yes let's keep it this way.